pysb/tools/sensitivity_analysis.py

```diff
@@ -0,0 +1,359 @@
+"""
+Sensitivity analysis tools created to demonstrate GPU powered
+analysis of PySB models.
+Information can be found in Harris et. al, Bioinformatics.
+Code written by James C. Pino
+
+"""
+import os
+from itertools import product
+import collections
+import matplotlib
+from matplotlib import gridspec, pyplot
+import numpy as np
+import pysb.integrate
+from pysb.bng import generate_equations
+plt = matplotlib.pyplot
+
+
+class InitialConcentrationSensitivityAnalysis:
+    """ Performs pairwise sensitivity analysis of initial conditions
+
+    """
+
+    def __init__(self, model, time_span, values_to_sample, objective_function,
+                 observable):
+        self.model = model
+        self.sim_time = time_span
+        generate_equations(self.model)
+        self.proteins_of_interest = list(
+                (i[1].name for i in model.initial_conditions))
+        if '__source_0' in self.proteins_of_interest:
+            self.proteins_of_interest.remove('__source_0')
+        self.proteins_of_interest = sorted(self.proteins_of_interest)
+        self.n_proteins = len(self.proteins_of_interest)
+        self.values_to_sample = values_to_sample
+        self.n_sam = len(self.values_to_sample)
+        self.b_matrix = []
+        self.b_prime_matrix = []
+        self.nm = self.n_proteins * self.n_sam
+        self.size_of_matrix = self.nm ** 2
+        self.shape_of_matrix = (self.nm, self.nm)
+        self.initial_conditions = np.zeros(len(self.model.species))
+        self.index_of_species_of_interest = self.create_index_of_species()
+        self.setup_sampling_matrix()
+        self.simulations = self._find_redundant_simulations()
+        self.objective_function = objective_function
+        self.observable = observable
+        self.standard = None
+        self.p_prime_matrix = np.zeros(self.size_of_matrix)
+        self.p_matrix = np.zeros(self.size_of_matrix)
+
+    def _calculate_objective(self, species):
+        """ Calculates fraction of change between standard value and newly obtained value
+
+        :param species:
+        :return: fraction of change, type float
+        """
+        return (self.objective_function(
+                species) - self.standard) / self.standard * 100.
+
+    def run(self, run_solver, save_name, out_dir):
```

Comment:
The `run` method (and any other public methods) should have a complete docstring, which follows the "napoleon" format used by PySB. The `save_name` and `out_dir` arguments should probably be optional, if saving files is not desired (e.g. for unit tests). A working "examples" section using one of PySB's example models would be ideal.